ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java

```diff
@@ -195,6 +196,23 @@ public void shouldHandleMultiDimensionalArray() throws Exception {
         assertThat(result, equalTo("item_11"));
     }
 
+    @Test
+    public void shouldHandleMultiDimensionalArrayWithBaseStartingFrom1() throws Exception {
+        final KsqlConfig localKsqlConfig =
+            new KsqlConfig(Collections.singletonMap(KsqlConfig.KSQL_FUNCTIONS_ARRAY_LEGACY_BASE_CONFIG, false));
+        final CodeGenRunner localCodeGenRunner = new CodeGenRunner(schema, localKsqlConfig, functionRegistry);
+        final String simpleQuery = "SELECT col14[1][1] FROM CODEGEN_TEST;";
+        final Analysis analysis = analyzeQuery(simpleQuery);
+        final ExpressionMetadata expressionEvaluatorMetadata = localCodeGenRunner.buildCodeGenFromParseTree
+            (analysis.getSelectExpressions().get(0));
+        final List<String> innerArray1 = Arrays.asList("item_11", "item_12");
+        final List<String> innerArray2 = Arrays.asList("item_21", "item_22");
+        final Object[] args = new Object[]{Arrays.asList(innerArray1, innerArray2)};
+        final Object result = expressionEvaluatorMetadata.getExpressionEvaluator().evaluate(args);
```

Comment:
consider refactoring this to be more like:

```
@Test
    public void shouldHandleMultiDimensionalArrayWithBaseStartingFrom1() throws Exception {
        // Given:
        final String simpleQuery = "SELECT col14[1][1] FROM CODEGEN_TEST;";

        final Object[] arguments = {Arrays.asList(
            Arrays.asList("item_11", "item_12"),
            Arrays.asList("item_21", "item_22"))};

        EasyMock.expect(ksqlConfig.getBoolean(KsqlConfig.KSQL_FUNCTIONS_ARRAY_LEGACY_BASE_CONFIG)).andReturn(false);
        EasyMock.replay(ksqlConfig);

        // When:
        
        final Object result = executeExpression(simpleQuery, arguments);

        // Then:
        assertThat(result, instanceOf(String.class));
        assertThat(result, equalTo("item_11"));
    }

    private Object executeExpression(final String simpleQuery, final Object[] arguments) throws Exception {
        final Analysis analysis = analyzeQuery(simpleQuery);
        final ExpressionMetadata expressionEvaluatorMetadata = codeGenRunner.buildCodeGenFromParseTree
            (analysis.getSelectExpressions().get(0));
        return expressionEvaluatorMetadata.getExpressionEvaluator().evaluate(arguments);
    }
```

To make the test more readable. The same could be applied to other tests, like the one above this one, for example.